pyproject.toml

```diff
@@ -1,3 +1,60 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = ["setuptools"]
```

Comment:
setuptools >= 64 is required for editable installs (pip install -e) with pyproject.toml-only builds. I don't think we need to encode that requirement here, just wanted to mention it.

Reply:
OK, this is quite odd. I tried changing our CI to do a non-editable pip install of the package, and I started seeing failures in `test_single_writer_multiple_reader_same_instance`. [Here's one such example](https://github.com/glotzerlab/signac/actions/runs/3726228439/jobs/6320325740). The problem seems isolated to specific combinations in the test matrix, so I suspect that the issue is in some invalid assumption of the test. It's completely unrelated to this PR's changes, so I just bumped the setuptools requirement to move forward and continue doing an editable install. Just wanted to document the issue here in case someone sees it elsewhere.